src/sample_package/test/my_node_rostest.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Created By: Valerian Ratu
+ * Created On: January 29, 2017
+ * Description: Integration testing for MyNode
+ */
+
+
+#include <MyNode.h>
+#include <gtest/gtest.h>
+#include <ros/ros.h>
+
+
+/**
+ * This is the helper class which will publish and subscribe messages which will test the node being instantiated
+ * It contains on the minimum:
+ *      publisher - publishes the input to the node
+ *      subscriber - publishes the output of the node
+ *      callback function - the callback function which corresponds to the subscriber
+ *      getter function - to provide a way for gtest to check for equality of the message recieved
+ */
+class MyClassTest : public testing::Test{
+protected:
+    virtual void SetUp(){
+        testPublisher = nh_.advertise<std_msgs::String>("subscribe_topic", 1);
+        testSubscriber = nh_.subscribe("/my_node/publish_topic", 1, &MyClassTest::callback, this);
+
+        // Let the publishers and subscribers set itself up timely
+        ros::Rate loop_rate(1);
+        loop_rate.sleep();
+    }
+
+    ros::NodeHandle nh_;
+    std::string messageOutput;
+    ros::Publisher testPublisher;
+    ros::Subscriber testSubscriber;
+
+public:
+
+    void callback(const std_msgs::String::ConstPtr msg){
+        messageOutput = msg->data.c_str();
+    }
+
+    std::string getMessage(){
+        return messageOutput;
+    }
+};
+
+TEST_F(MyClassTest, MyNodeTest){
```

Comment:
I'm thinking that perhaps we should name the test class `MyNodeTest`, with the second name being reserved to indicate very briefly the functionality you're trying to test.